File: modules/seamless-recorder/ios/SeamlessRecorderModule.swift
import ExpoModulesCore

public class SeamlessRecorderModule: Module {
    public func definition() -> ModuleDefinition {
        Name("ExpoSeamlessRecorder")

        View(SeamlessRecorderView.self) {
            Events("onChunkReady", "onError")

            Prop("facing") { (view: SeamlessRecorderView, facing: String?) in
                view.setFacing(facing ?? "front")
            }

            Prop("videoQuality") { (view: SeamlessRecorderView, quality: String?) in
                view.setQuality(quality ?? "480p")
            }

            AsyncFunction("startRecording") { (view: SeamlessRecorderView, promise: Promise) in
                view.startRecording(promise: promise)
            }.runOnQueue(.main)

            AsyncFunction("stopRecording") { (view: SeamlessRecorderView, promise: Promise) in
                view.stopRecording(promise: promise)
            }.runOnQueue(.main)
        }
    }
}

File: modules/seamless-recorder/android/src/main/java/expo/modules/seamlessrecorder/SeamlessRecorderModule.kt
package expo.modules.seamlessrecorder

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import expo.modules.kotlin.Promise
import expo.modules.kotlin.functions.Queues

class SeamlessRecorderModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoSeamlessRecorder")

    View(SeamlessRecorderView::class) {
      Events("onChunkReady", "onError")

      Prop("facing") { view: SeamlessRecorderView, facing: String ->
        view.setFacing(facing)
      }

      Prop("videoQuality") { view: SeamlessRecorderView, quality: String ->
        view.setQuality(quality)
      }

      Prop("sessionId") { view: SeamlessRecorderView, sessionId: String ->
        view.setSessionId(sessionId)
      }

      AsyncFunction("startRecording") { view: SeamlessRecorderView, promise: Promise ->
        view.startRecording(promise)
      }.runOnQueue(Queues.MAIN)

      AsyncFunction("stopRecording") { view: SeamlessRecorderView, promise: Promise ->
        view.stopRecording(promise)
      }.runOnQueue(Queues.MAIN)
    }
  }
}
